sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonParser.scala

```diff
@@ -361,6 +361,12 @@ class JacksonParser(
         // For such records, all fields other than the field configured by
         // `columnNameOfCorruptRecord` are set to `null`.
         throw BadRecordException(() => recordLiteral(record), () => None, e)
+      case e: CharConversionException if options.encoding.isEmpty =>
+        val msg =
+          """JSON parser cannot handle a character in its input.
+            |Specifying encoding as an input option explicitly might help to resolve the issue.
+            |""".stripMargin + e.getMessage
+        throw new CharConversionException(msg)
```

Comment:
This will lose the original stack trace, we should do `throw BadRecordException(() => recordLiteral(record), () => None, e)`

Reply:
BTW we should also follow the existing rule and wrap the exception with `BadRecordException`. See the code above.